source/common/upstream/upstream_impl.cc

```diff
@@ -1726,15 +1740,24 @@ bool BaseDynamicClusterImpl::updateDynamicHostList(
   if (!current_priority_hosts.empty() && dont_remove_healthy_hosts) {
     erase_from =
         std::remove_if(current_priority_hosts.begin(), current_priority_hosts.end(),
-                       [&all_new_hosts, &new_hosts_for_current_priority, &final_hosts,
+                       [&all_new_hosts, &new_hosts_for_current_priority,
+                        &existing_hosts_for_current_priority_locality_changed, &final_hosts,
                         &max_host_weight](const HostSharedPtr& p) {
+                         // This host is being added as a new host, total replacement
+                         // remove the older host
```

Comment:
This comment seems a bit incomplete, could you elaborate? Are we adding a new host with the updated locality, then remove the old one?

Reply:
updated